### ISSUE TITLE
Key value integration versioning

### ISSUE THREAD

**Issue by Boolos** [Shippable/support#4346]
### Question about key-vaule integration versioning:
Is there anyway to trigger a new build when you update the value in a key-value integration? Similar to the way the Params resource works?

I'm using a key-value integration to store secrets that generate a kube Secrets object, prior to deploying. similar to this article: http://blog.shippable.com/kubernetes-tutorial-using-secretes-in-your-kubernetes-application

It works fine from a deployment standpoint. the only problem is that if i go back to the Shippable UI and update one of the values, it doesn't trigger a new version of the Key-Value integration. So when the pipeline subsequently runs, the Deploy does not detect that anything has changed.  I'm guessing I could do a job trigger or update job state myself by changing the `versionName`, but I was hoping the key-value integration would work similarly to the Params resource and automatically update the versionTemplate upon changing values via the UI.

**Comment by trriplejay** [comment]
Even if the integration resource were versioned based on your changing your resource key/value, in this case the deploy job would trigger, but would still not deploy the manifests, because the manifests themselves would not have changed.  Manifests don't directly utilize key/value integrations, or integration resources, and when the deploy job looks for changes, it looks at each individual manifest.  The manifest is made up of images, params, and dockerOptions, so that's why a changed parameter will result in a deployment, while a changed integration will not.

I'd recommend making some kind of change to a params or dockerOptions resource.  An alternative is to use a trigger resource combined with the `force: true` modifier on the IN step of the job.  This means that that particular manifest (or manifests if the IN is another deploy job) will always be deployed when the deploy job runs, even if the manifest hasn't changed.

Does that make sense? Let me know if you have more questions.

**Comment by Boolos** [comment]
Yup that makes sense, thanks.  Are there any plans to make Manifests directly utilize key/value integrations?  That seems to be the best way to store secrets (as storing them in the Params resource after hashing them through the Shippable UI is a bit cumbersome), and it would be nice to be able to version both the params and the secrets for a service together in order to create reproducible deployments without having to rely on some extraneous resource (e.g. a trigger) in order to figure out what version of secrets were deployed with a give manifest version.
